src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -51,6 +60,50 @@ object EntityClient {
     }
   }
 
+  //returns (contents of participants.tsv, contents of samples.tsv)
+  def unzipTSVs(bagName: String, zipFile: ZipFile)(op: (Option[String], Option[String]) => Future[PerRequestMessage]): Future[PerRequestMessage] = {
+    val zipEntries = zipFile.entries.asScala
+
+    val rand = java.util.UUID.randomUUID.toString.take(8)
+    val participantsTmp = File.createTempFile(s"$rand-participants", ".tsv")
+    val samplesTmp = File.createTempFile(s"$rand-samples", ".tsv")
+
+    val unzippedFiles = zipEntries.foldLeft((None: Option[String], None: Option[String])){ (acc: (Option[String], Option[String]), ent: ZipEntry) =>
+      if(!ent.isDirectory && (ent.getName.contains("/participants.tsv") || ent.getName.equals("participants.tsv"))) {
+        acc._1 match {
+          case Some(x) => throw new FireCloudException(s"More than one participants.tsv file found in BDBag $bagName")
```

Comment:
nit: bagit, not bdbag

Reply:
also: i assume this exception turns into a 400 somewhere upstream?

Reply:
no it does not, I should fix that